pkg/reconciler/buildtemplate/resources/imagecache.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/knative/pkg/kmeta"
+
+	"github.com/knative/build/pkg/apis/build/v1alpha1"
+	"github.com/knative/build/pkg/reconciler/buildtemplate/resources/names"
+	caching "github.com/knative/caching/pkg/apis/caching/v1alpha1"
+)
+
+// Note: namespace is passed separately because this may be used for
+// cluster-scoped stuff as well.
+func makeImageCachesFromSpec(
+	namespace string,
+	bt names.ImageCacheable,
+) []caching.Image {
+	var caches []caching.Image
+
+	// Avoid duplicates.
+	images := make(map[string]struct{})
+
+	for index, container := range bt.TemplateSpec().Steps {
+		if strings.Contains(container.Image, "$") {
```

Comment:
We can resolve this and assume the default at least.

Reply:
Discussed offline, and added a TODO for now.